src/Raven.Server/Documents/Sharding/Background/ShardedPeriodicDocumentsMigrator.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Server.Background;
+using Raven.Server.ServerWide.Commands.Sharding;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Logging;
+
+namespace Raven.Server.Documents.Sharding.Background
+{
+    public class ShardedPeriodicDocumentsMigrator : BackgroundWorkBase
+    {
+        private readonly ShardedDocumentDatabase _database;
+
+        public ShardedPeriodicDocumentsMigrator(ShardedDocumentDatabase database) : base(database.ShardedDatabaseName, database.ShardedDocumentsStorage.DocumentDatabase.DatabaseShutdown)
+        {
+            _database = database;
+        }
+
+        protected override async Task DoWork()
+        {
+            await WaitOrThrowOperationCanceled(_database.Configuration.Sharding.PeriodicDocumentsMigrationInterval.AsTimeSpan);
+
+            while (_database.ServerStore.Sharding.HasActiveMigrations(_database.ShardedDatabaseName))
+                await WaitOrThrowOperationCanceled(TimeSpan.FromMilliseconds(300));
+
+            await ExecuteMoveDocuments();
+        }
+
+        internal async Task ExecuteMoveDocuments()
+        {
+            try
+            {
+                int bucket = -1;
+                int moveToShard = -1;
+                bool found = false;
+                using (_database.ShardedDocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var configuration = _database.ShardingConfiguration;
+                    for (var index = 0; index < configuration.BucketRanges.Count; index++)
+                    {
+                        var range = configuration.BucketRanges[index];
+                        if (range.ShardNumber == _database.ShardNumber)
+                            continue;
+
+                        var start = range.BucketRangeStart;
+                        var end = index == configuration.BucketRanges.Count - 1
```

Comment:
What about prefix ranges?

Reply:
it is working (it is part of the `BucketRanges`, see for example `PeriodicDocumentsMigratorShouldWork_MultipleWrongBuckets` test)